src/components/DayPicker.jsx

```diff
@@ -399,19 +412,27 @@ class DayPicker extends React.Component {
   }
 
   onPrevMonthClick(nextFocusedDate, e) {
-    const { numberOfMonths, isRTL } = this.props;
-    const { calendarMonthWidth } = this.state;
+    const { daySize, isRTL } = this.props;
+    const { calendarMonthWidth, currentMonth, monthTitleHeight } = this.state;
 
     if (e) e.preventDefault();
 
-    let translationValue = this.isVertical() ? this.calendarMonthHeights[0] : calendarMonthWidth;
+    const newMonth = currentMonth.clone().subtract(1, 'month');
+    const numberOfWeeks = getNumberOfCalendarMonthWeeks(newMonth, this.getFirstDayOfWeek());
```

Comment:
I mostly think it would be cool to reduce the work done on prev/next click. Wondering if we can pre-calculate and key the months somehow, or even memoize `getNumberOfCalendarMonthWeeks` or something so that we don't have to recalculate week count and shift `calendarMonthWeeks` on each transition.